File: state.py
import time

class State():
    """
    A sate class (that should be an Enum) in both mobile/base stations, will handle timing and state change
    """

    # common
    IDLE = 0

    # Base station States
    SIFS_before_emitCTS = 1
    emitCTS = 2
    SIFS_before_rcvPKT = 3
    rcvPKT = 4
    SIFS_before_emitACK = 5
    emitACK = 6

    # Mobile Station States
    DIFS_before_Countdown = 7
    Countdown = 8
    emitRTS = 9
    SIFS_before_rcvCTS = 10 
    rcvCTS = 11
    SIFS_before_emitPKT = 12 
    emitPKT = 13
    SIFS_before_rcvACK = 14 
    rcvACK = 15

    def __init__(self, state = IDLE):
        self.changeTo(state)

    def changeTo(self, newState):
        self.ticks = time.time()
        self.state = newState

    def elapsedTimeSinceChange(self, i):
        return (time.time() - self.ticks) > i



if __name__ == "__main__":
    s = State()
    print(s.state)

File: java implementation/State.java
import java.util.HashMap;

public enum State {
    IDLE,
    SIFS_before_emitCTS,
    emitCTS,
    SIFS_before_rcvPKT,
    rcvPKT,
    SIFS_before_emitACK,
    emitACK,
    DIFS_before_Countdown,
    Countdown,
    emitRTS,
    SIFS_before_rcvCTS,
    rcvCTS,
    SIFS_before_emitPKT,
    emitPKT,
    SIFS_before_rcvACK,
    rcvACK;

    private final HashMap<String, String> stateColor = new HashMap<String, String>();

    State(HashMap<String, String> stateColor) {
        stateColor.put("no Frame","White");
        stateColor.put("RTS","Bright Blue");
        stateColor.put("CTS","Dark Blue");
        stateColor.put("Data Frame","Black");
        stateColor.put("ACK","Green");
        stateColor.put("Channel Free","Grey");
        stateColor.put("Channel Busy","Red");
        stateColor.put("Channel Considered Busy","Yellow");
        stateColor.put("Countdown","Dark Grey");
    }

    State() {
    }
    public void getStateColor(String key)
    {
        stateColor.get(key);
    }

}

